pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -336,11 +336,39 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// Remove dead container log symlinks.
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
-	for _, logSymlink := range logSymlinks {
-		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
-			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+	if cgc.manager.runtimeName == "docker" {
+		for _, logSymlink := range logSymlinks {
+			if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+				firstLinkDstPath, err := os.Readlink(logSymlink)
+				if err != nil {
+					glog.Errorf("Failed to resolve legacy container log symlink %q: %v", logSymlink, err)
+					return nil
+				}
+				secondLinkDstPath, err := os.Readlink(firstLinkDstPath)
+				if err != nil {
+					glog.Errorf("Failed to resolve container log symlink %q: %v", firstLinkDstPath, err)
+					return nil
```

Comment:
continue instead of return?

Reply:
@feiskyer Oh,yes sir."continue" is much more suitable,thank you so much.